src/caughtup/__tests__/caughtUpReducer-test.js

```diff
@@ -15,41 +15,34 @@ import {
 import { objectFromEntries } from '../../jsBackport';
 
 describe('caughtUpReducer', () => {
+  test('RESET_ACCOUNT_DATA', () => {
+    expect(
+      [
+        { ...eg.action.message_fetch_complete, foundNewest: true, foundOldest: true },
+        eg.action.reset_account_data,
+      ].reduce(caughtUpReducer, eg.baseReduxState.caughtUp),
+    ).toEqual(eg.baseReduxState.caughtUp);
+  });
```

Comment:
Hmm. I like the idea of using actions and the reducer to construct the starting state. But I find this specific way of writing the test a bit unsatisfying because
* for the test to be effective (that is, for it to not just vacuously pass even if we were to break the thing it's testing), it's essential that the first action in this list produces some state that *isn't* equal to the base/initial state;
* but it's not immediately obvious on reading the test that that's true — it depends on some details of what's in that action, and one has to read it closely and think through the meaning of `caughtUp` and how it should behave here to confirm that it's true;
* relatedly, if we were to make some future change to the reducer's semantics such that this first action *didn't* produce anything different from the initial state — not super likely in this example, but again to reach that conclusion one has to think about the semantics of `caughtUp` and decide that this is fundamental to it — then the test would become vacuous, and nothing would cause us to closely reread this code and notice that.

Here's an alternative version, which preserves the property that we construct the state through the reducer rather than as an object literal:
```js
  test('RESET_ACCOUNT_DATA', () => {
    const initialState = eg.baseReduxState.caughtUp;
    const prevState = caughtUpReducer(initialState, {
      ...eg.action.message_fetch_complete,
      foundNewest: true,
      foundOldest: true,
    });
    expect(prevState).not.toEqual(initialState);

    expect(caughtUpReducer(prevState, eg.action.reset_account_data)).toEqual(initialState);
  });
```

That way, the setup explicitly checks that the state before reset isn't already equal to the initial state.

(Similarly for several other reducers' tests added in this commit.)

Reply:
Ah, yeah, makes sense. Thanks!